pkg/arvo/sur/btc-node-hook.hoon

```diff
@@ -0,0 +1,2489 @@
+      $:  name=?(%ipv4 %ipv6 %onion)
+          limited=?
+          reachable=?
+          proxy=@t
+          proxy-randomize-credentials=?
+      ==
+    ::  $local-address:
+    ::
+    ::  Used in:
+    ::      - %get-network-info
+    ::
+    +$  local-address  [address=@t port=@ud score=@ud]
+    ::  $node-address-info:
+    ::
+    ::  Used in:
+    ::      - %get-node-addresses
+    ::
+    +$  node-address-info  [time=@da services=@ud address=@t port=@ud]
+    ::  $peer-info:
+    ::
+    ::  Used in:
+    ::      - %get-peer-info
+    ::
+    +$  peer-info
+      $:  id=@ud
+          addr=@t
+          addr-bind=@t
+          addr-local=(unit @t)
+          services=@t
+          relay-txes=?
+          last-send=@da
+          last-recv=@da
+          bytes-sent=@ud
+          bytes-recv=@ud
+          conn-time=@da
+          time-offset=@ud
+          ping-time=@rd
+          min-ping=@rd
+          ping-wait=(unit @ud)
+          version=@ud
+          subver=@t
+          inbound=?
+          addnode=?
+          starting-height=@ud
+          ban-score=@ud
+          synced-headers=@ud
+          synced-blocks=@ud
+          inflight=(list @ud)
+          whitelisted=?
+          min-fee-filter=@rd
+          bytes-sent-per-msg=(map @t @ud)
+          bytes-recv-per-msg=(map @t @ud)
+      ==
+    ::  $banned:
+    ::
+    ::    Used in:
+    ::      - %list-banned
+    ::
+    +$  banned  [address=@t banned-until=@da ban-created=@da ban-reason=@t]
+
+    ::  $chain-status:
+    ::
+    ::    Used in:
+    ::      - %get-chain-tips
+    ::
+    +$  chain-status
+      $?  %invalid
+          %headers-only
+          %valid-headers
+          %valid-fork
+          %active
+      ==
+    ::  $soft-fork-status:
+    ::
+    ::    Used in:
+    ::      - %get-blockchain-info/$bip9-softforks
+    ::
+    +$  soft-fork-status  ?(%defined %started %'locked_in' %active %failed)
+    ::  $purpose:
+    ::
+    ::    Used in:
+    ::      - %get-addresses-by-label
+    ::      - %get-address-info
+    ::
+    +$  purpose  ?(%send %receive)
+    ::  $address-type:
+    ::
+    ::    Used in:
+    ::      - %add-multisig-address
+    ::
+    +$  address-type  ?(%legacy %p2sh-segwit %bech32)
+    ::  $bip125-replaceable:
+    ::
+    ::    Used in:
+    ::      - %get-transaction
+    ::      - %list-transactions
+    ::      - $tx-in-block
+    ::
+    +$  bip125-replaceable  ?(%yes %no %unknown)
+    ::  $network-name:
+    ::
+    ::    Used in:
+    ::      - %get-blockchain-info
+    ::
+    +$  network-name  ?(%main %test %regtest)
+    ::  $sig-hash:
+    ::
+    ::    Used in:
+    ::      - %wallet-process-psbt
+    ::      - %sign-raw-transaction-with-wallet
+    ::      - %sign-raw-transaction-with-key
+    ::      - %decode-psbt
+    ::
+    +$  sig-hash
+      $?  %'NONE'
+          %'SINGLE'
+          %'ALL|ANYONECANPAY'
+          %'NONE|ANYONECANPAY'
+          %'SINGLE|ANYONECANPAY'
+          %'ALL'
+      ==
```

Comment:
I think that the RPC API[1] requieres the type of the signature to be send exactly like that. We really don't need for the type to match it but it was just to avoid having a conversion functions

- [1] https://bitcoincore.org/en/doc/0.19.0/rpc/rawtransactions/signrawtransactionwithkey/